### PR TITLE
[RLlib] Issue 21297: Ignore PPO KL-loss term completely if kl-coeff == 0.0 to avoid NaN values due to some discrete action probs==0.0

### PR DESCRIPTION
**Pull request by sven1977**
Issue 21297: Ignore PPO KL-loss term completely if kl-coeff == 0.0 to avoid NaN values due to some discrete action probs==0.0

## Why are these changes needed?

https://github.com/ray-project/ray/issues/21297

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/ppo/ppo_tf_policy.py [hunk 1]
```diff
@@ -110,9 +115,12 @@ def reduce_mean_valid(t):
         vf_loss = mean_vf_loss = tf.constant(0.0)
 
     total_loss = reduce_mean_valid(-surrogate_loss +
-                                   policy.kl_coeff * action_kl +
                                    policy.config["vf_loss_coeff"] * vf_loss -
                                    policy.entropy_coeff * curr_entropy)
+    # Add mean_kl_loss (already processed through `reduce_mean_valid`),
+    # if necessary.
+    if policy.config["kl_coeff"] > 0.0:
```
Comment (gjoliver) [inline rllib/agents/ppo/ppo_tf_policy.py:122]: seems this if statement is not necessary, 0.0 case, mean_kl_loss is zero. same below.

**Review by gjoliver — APPROVED** [review]
just a minor comment.
thanks.

File: rllib/agents/ppo/ppo_tf_policy.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/ppo/ppo_tf_policy.py:122]: Normally, yes, but if the calculated kl is Inf (due to some actions having almost 0.0 probability -> log(a) = -inf), then multiplying it with 0.0 will result in NaN. :)
